dotCMS/src/main/java/com/dotcms/rest/api/v2/contenttype/FieldResource.java

```diff
@@ -226,15 +220,17 @@ public Response updateContentTypeFieldById(@PathParam("typeId") final String typ
 
             } else {
 
-                Field currentField = fapi.find(fieldId);
+                final Field currentField = fieldAPI.find(fieldId);
 
                 if (!currentField.id().equals(field.id())) {
 
-                    response = ExceptionMapperUtil.createResponse(null, "Field id '"+ fieldId +"' does not match a field with id '"+ field.id() +"'");
+                    throw new DotDataValidationException("Field id '"+ fieldId +"' does not match a field with id '"+ field.id() +"'");
 
+                } else if(!currentField.variable().equals(field.variable())){
```

Comment:
Let's move this business rule to the api level. It also applies when using the java api directly